recipes-kernel/linux/files/patches-5.10/0119-arm64-dts-ti-iot2050-add-the-support-for-the-m.2-var.patch

```diff
@@ -0,0 +1,311 @@
+From 8d2493c70af10e774f99fe00a56084e9e32943e6 Mon Sep 17 00:00:00 2001
+From: chao zeng <chao.zeng@siemens.com>
+Date: Tue, 11 Jan 2022 11:28:24 +0800
+Subject: [PATCH 119/123] arm64: dts :ti: iot2050: add the support for the m.2
+ variant
+
+1.configure the m2 board dts based on the pg2 advanced variant
+2.write the overlay for different lane combination
+
+Signed-off-by: chao zeng <chao.zeng@siemens.com>
+---
+ arch/arm64/boot/dts/ti/Makefile               |  4 +
+ .../dts/ti/k3-am65-iot2050-common-pg2-m2.dtsi | 33 ++++++++
+ .../dts/ti/k3-am65-iot2050-common-pg2.dtsi    | 23 +-----
+ .../k3-am654-iot2050-bkey-pcie-ekey-pcie.dts  | 29 +++++++
+ .../dts/ti/k3-am654-iot2050-bkey-pciex2.dts   | 29 +++++++
+ .../k3-am654-iot2050-bkey-usb3-ekey-pcie.dts  | 32 ++++++++
+ .../dts/ti/k3-am6548-iot2050-advanced-m2.dts  | 80 +++++++++++++++++++
+ 7 files changed, 208 insertions(+), 22 deletions(-)
+ create mode 100644 arch/arm64/boot/dts/ti/k3-am65-iot2050-common-pg2-m2.dtsi
+ create mode 100644 arch/arm64/boot/dts/ti/k3-am654-iot2050-bkey-pcie-ekey-pcie.dts
+ create mode 100644 arch/arm64/boot/dts/ti/k3-am654-iot2050-bkey-pciex2.dts
+ create mode 100644 arch/arm64/boot/dts/ti/k3-am654-iot2050-bkey-usb3-ekey-pcie.dts
+ create mode 100644 arch/arm64/boot/dts/ti/k3-am6548-iot2050-advanced-m2.dts
+
+diff --git a/arch/arm64/boot/dts/ti/Makefile b/arch/arm64/boot/dts/ti/Makefile
+index e8a07d411627..3de28e26aa94 100644
+--- a/arch/arm64/boot/dts/ti/Makefile
++++ b/arch/arm64/boot/dts/ti/Makefile
+@@ -11,6 +11,10 @@ dtb-$(CONFIG_ARCH_K3) += k3-am6528-iot2050-basic.dtb
+ dtb-$(CONFIG_ARCH_K3) += k3-am6528-iot2050-basic-pg2.dtb
+ dtb-$(CONFIG_ARCH_K3) += k3-am6548-iot2050-advanced.dtb
+ dtb-$(CONFIG_ARCH_K3) += k3-am6548-iot2050-advanced-pg2.dtb
++dtb-$(CONFIG_ARCH_K3) += k3-am6548-iot2050-advanced-m2.dtb
++dtb-$(CONFIG_ARCH_K3) += k3-am654-iot2050-bkey-pciex2.dtbo
++dtb-$(CONFIG_ARCH_K3) += k3-am654-iot2050-bkey-usb3-ekey-pcie.dtbo
++dtb-$(CONFIG_ARCH_K3) += k3-am654-iot2050-bkey-pcie-ekey-pcie.dtbo
+ 
+ dtb-$(CONFIG_ARCH_K3) += k3-j721e-common-proc-board.dtb
+ 
+diff --git a/arch/arm64/boot/dts/ti/k3-am65-iot2050-common-pg2-m2.dtsi b/arch/arm64/boot/dts/ti/k3-am65-iot2050-common-pg2-m2.dtsi
+new file mode 100644
+index 000000000000..40ad18747d95
+--- /dev/null
++++ b/arch/arm64/boot/dts/ti/k3-am65-iot2050-common-pg2-m2.dtsi
```

Comment:
Actually, this file is now true "common-pg2". The file where the bits came from becomes kind of "basic-advanced-pg2", ie. without m2.

Reply:
...or what is the naming convention? Does "PG2" only stand for Basic/Advanced, and M2 is a separate name? Still trying to learn the scheme.

Reply:
We'll have a good discussion of naming conventions